src/parse.rs

```diff
@@ -530,6 +533,43 @@ impl<'input, 'callback> Parser<'input, 'callback> {
                         }
                     }
                 }
+                ItemBody::MaybeMath => {
+                    let next = self.tree[cur_ix].next;
+                    let start_ix = if let Some(next_ix) = next {
+                        self.tree[next_ix].item.start
+                    } else {
+                        self.tree[cur_ix].item.body = ItemBody::Text;
+                        prev = cur;
+                        cur = self.tree[cur_ix].next;
+                        continue;
+                    };
+
+                    let end_ix =
+                        if let Some(i) = scan_math_inline(&block_text.as_bytes()[start_ix..]) {
+                            start_ix + i
+                        } else {
+                            self.tree[cur_ix].item.body = ItemBody::Text;
+                            prev = cur;
+                            cur = self.tree[cur_ix].next;
+                            continue;
+                        };
+
+                    let node = scan_nodes_to_ix(&self.tree, next, end_ix + 1);
+                    if end_ix > start_ix {
+                        let cow_ix = self
+                            .allocs
+                            .allocate_cow(block_text[start_ix + 1..end_ix].into());
```

Comment:
```suggestion
                            .allocate_cow(block_text[start_ix..end_ix].into());
```

Off-by-one? The first character of inline math seems to get dropped without this suggestion.